_posts/2024-08-23-gsoc24-final-report.md

```diff
@@ -0,0 +1,133 @@
+[shUnit2](https://github.com/kward/shunit2) framework to organize the tests,
+along with specific scripts to monitor and validate CPU usage with the
+`--cpu-scaling` option. This allowed `kw build` to be tested without the need
+to compile the entire kernel
+
+The integration tests for the `kw ssh` feature were more complex due to the use
+of nested containers. Adapting the tests for this scenario was challenging, but
+it also provided a valuable learning opportunity. Additionally, I had to manage
+the execution of commands with special characters inside the containers. To
+address this, I developed functions to ensure that these commands were
+correctly interpreted by the shell.
+
+I faced several challenges during my Google Summer of Code journey, but they
+were overcome with the support of mentors who were always available to answer
+my questions.
+
+# Blogpost Series Timeline
+
+To provide a detailed overview of the work done during the Google Summer of
+Code, I have prepared a series of blog posts that explore different aspects of
+the **kworkflow** project. Here is a timeline of the posts, with direct links to
+each one:
+
+1. [Accepted to Google Summer of Code]({{site.url}}/got-accepted-into-gsoc/)
+
+2. [Introduction to Integration Testing in kworkflow ]({{site.url}}/introduction-to-integration-testing/)
+
+3. [Integration Testing for kw ssh]({{site.url}}/integration-for-kw-ssh/)
+
+4. [Integration Testing for kw build]({{site.url}}/integration-for-kw-build/)
+
+# Contributions
+
+Throughout the project, I created several pull requests (PRs) addressing
+different aspects of kworkflow. Each PR was carefully crafted to enhance
+functionality, increase test coverage, and ensure code robustness. Below are
+some of the most significant PRs:
+
+| Pull Request | N° of Commits | 
+|--------------|:---------------:|
+| [setup: install kernel build dependencies](https://github.com/kworkflow/kworkflow/pull/1108) |4|
+| [tests: integration: device_test: modify kw device integration test to run entirely in container](https://github.com/kworkflow/kworkflow/pull/1135) | 2 |
+| [tests: integration: refactor kw_version_test to run entirely in container](https://github.com/kworkflow/kworkflow/pull/1113) | 1 |
+| [run_tests: dedicate a container per test file for integration tests](https://github.com/kworkflow/kworkflow/pull/1130) | 2 |
+| [run_tests: streamline test execution logic](https://github.com/kworkflow/kworkflow/pull/1148) | 1 |
+| [tests: integration: self_update_test: add self-update test](https://github.com/kworkflow/kworkflow/pull/1055) | 3 |
+| [tests: integration: deploy_test: introducing deploy tests](https://github.com/kworkflow/kworkflow/pull/1161) | 2 |
+| [tests: integration: kw_ssh_test: Add integration tests for kw ssh functionality](https://github.com/kworkflow/kworkflow/pull/1116) | 5 |
+| [tests: integration: build_test: add the kw build test](https://github.com/kworkflow/kworkflow/pull/1143) | 3 |
+
```

Comment:
Maybe it is worth mentioning that even though these PRs are your palpable contributions, much work was done "offline" because you had to research the best course of action, communicate it with the mentors, collect feedback, and so on.